### PR TITLE
foundation work on Theming + style cleanup

### PR DESCRIPTION
**Pull request by Simon-Laux**
-> anything that is already in styled components can use themed color vars
now
+some scss cleanup

Just laying some foundation for theming,
Momentary you can set your theme over in the dev console:
```
Overwrite your current theme:
window.ThemeManager.setTheme(theme_object)
Reset your active Theme:
window.ThemeManager.setTheme({})
Get current Theme:
window.ThemeManager.getCurrentTheme()
```
It's automatically saved in local storage on change.

### CODE REVIEW

File: src/renderer/Theming.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  constructor () {
+    super()
+    const themeJSON = window.localStorage.getItem('theme')
+    this.currentTheme = themeJSON !== null ? JSON.parse(themeJSON) : {}
+
+    window.ThemeManager = this // only for using fron the dev console
+  }
+
+  setTheme (theme) {
+    window.localStorage.setItem('theme', JSON.stringify(theme))
```
Comment (Jikstra) [inline src/renderer/Theming.js]: Can we use rc for this? This way you can set the theme via the config.ini file

File: src/renderer/Theming.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    this.state = {
+      theme: manager.getCurrentTheme()
+    }
+    this.update = this.update.bind(this)
+  }
+
+  componentDidMount () {
+    manager.addListener('update', this.update)
+  }
+
```
Comment (Jikstra) [inline src/renderer/Theming.js]: Does it really need this much? Isn't something like (just an idea, no actual code) enough:
```
const lightTheme = {...}
const darkTheme = {...}

...
function ThemeProvider {
   const [selectedTheme, setSelectedTheme] = useState('light')
   dc.on('DC_SELECT_THEME', setSelectedTheme)
   return (<StyledThemeProvider theme={rc.theme === 'light' ? lightTheme : darkTheme}>
      {this.props.children}
    </StyledThemeProvider>)
}
```

File: src/renderer/Theming.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  deltaChatMessageBubbleSelf: '#efffde',
+  deltaChatMessageBubbleOther: '#ffffff',
+  deltaInfoMessageBubbleBg: '#000000',
+  deltaInfoMessageBubbleColor: 'white',
+  deltaChatMessageBubbleSelfStatusColor: '#4caf50',
+  deltaFocusBlue: '#42A5F5',
+  deltaSelected: '#4c6e7d' // same as deltaPrimaryBgLight
+})
+
+class ThemeManager extends EventEmitter {
```
Comment (Jikstra) [inline src/renderer/Theming.js]: At least I don't really understand why we need a ThemeManager/EventEmitter

**Comment by Jikstra** [comment]
Very cool that you're pushing Theming!

File: src/renderer/Theming.js (hunk 3 above)
Comment (Simon-Laux) [reply src/renderer/Theming.js]: Thats my quick 'n (maybe) dirty approach to make it changeable at runtime

File: src/renderer/Theming.js (hunk 2 above)
Comment (Simon-Laux) [reply src/renderer/Theming.js]: I'd like to have more than one theme / maybe custom themes. But doing it from the backend would be good for working with files.

File: src/renderer/Theming.js (hunk 1 above)
Comment (Simon-Laux) [reply src/renderer/Theming.js]: I like the idea to use a config file and/or theme file. What about a steam like aproach? having a theme folder where users can add theme files. that then get displayed in an theme dropdown in the settings. But on the other hand I like the option to create your theme from the the dev console so maybe a combination of both methods... Or just a load from file option in the front end like upload a `theme.yaml` file to import it. And download to export it...

Comment (Jikstra) [reply src/renderer/Theming.js]: I would prefer having it in the config folder and not in some browser/electron storage (as an upload/download would suggest). But for the moment it's not about making customized themes but integrating a dark theme.

Comment (Simon-Laux) [reply src/renderer/Theming.js]: > But for the moment it's not about making customized themes but integrating a dark theme.

Why not set the foundation now? we don't have to show the option to the user. I'll see in how far it makes sense. I mean we don't need a theme editor now for example. But a folder with theme files from that you get a drop down which the user can select from in the settings and maybe the option to change it from the dev console seems doable and not too much out of scope...

**Comment by Simon-Laux** [comment]
I have a few Ideas on how the theme can be loaded, we can hold an InRealLife/IRC meeting or a/v call about this topic. @Jikstra @MIntrovrt
My current Opinion:
- I like to have the possibility to change the current theme on the fly, like over the js api I implemented now or over watching and hot reloading the theme file (when watching make it an command line option.)
- I also like having the themes and which theme to use in files rather than local storage.

**Comment by Jikstra** [comment]
> I also like having the themes and which theme to use in files rather than local storage.

+1 we should put as less stuff in local storage as possible. The gold standard would be to move a `.config/DeltaChat` at a completely different system and being able to continue as nothing would have happend.

> I like to have the possibility to change the current theme on the fly, like over the js api I implemented now or over watching and hot reloading the theme file (when watching make it an command line option.)

Makes sense. Also i think it would be good to be able to add some global css to overwrite more stuff than just changing some color variables. But, maybe that's hard because of the random names of StyledComponents (that's a minus point for them :P)

**Comment by Simon-Laux** [comment]
`.config/DeltaChat` - the local storage is also saved there as far as I know

> to overwrite more stuff than just changing some color variables

thats the point where you probably also want to change some structure and some html so forking will be fine.
Something that we could provide after color theming and changing chat view background.. Is changing emoji-&normal fonts but to cram them into theme files we would need folders or zip archives so I would rather provide them all as separate options.

**Comment by Simon-Laux** [comment]
We could have 3 standard options in the menu:
`Use System style (dark or light)`
`Dark Theme`
`Light Theme`
And then all themes from the theme folders (one user folder in config and one in the deltachat program directory for more included ones)

**Comment by Simon-Laux** [comment]
related: #409, #457

**Comment by Simon-Laux** [comment]
As this cleans up more css and doesn't do anything by default, I suggest merging it noalso @Jikstra's Theming efforts can use the new variables so its easier to add theming later.
#889

**Pull request merged**